File: backend/app/schemas/project.py
from pydantic import BaseModel
from datetime import datetime


class ProjectBase(BaseModel):
    ProjectName: str
    Description: str
    CustomerID: int
    ManagerID: int
    Status: str
    StartDate: datetime
    EndDate: datetime


class ProjectCreate(ProjectBase):
    pass


class ProjectUpdate(ProjectBase):
    pass


class ProjectResponse(ProjectBase):
    ProjectID: int
    CreatedAt: datetime

    class Config:
        from_attributes = True

File: frontend/types/project.ts
export interface Project {
    ProjectID: number;
    ProjectName: string;
    Description: string;
    CustomerID: number;
    ManagerID: number;
    Status: string;
    StartDate: string;
    EndDate: string;
    CreatedAt: string;
  }
  
  export interface ProjectCreate {
    ProjectName: string;
    Description: string;
    CustomerID: number;
    ManagerID: number;
    Status: string;
    StartDate: string;
    EndDate: string;
  }
  
  export interface ProjectUpdate {
    ProjectName: string;
    Description: string;
    CustomerID: number;
    ManagerID: number;
    Status: string;
    StartDate: string;
    EndDate: string;
  }